app/src/main/java/de/westnordost/streetcomplete/data/upload/VersionIsBannedChecker.kt

```diff
@@ -1,29 +1,33 @@
 package de.westnordost.streetcomplete.data.upload
 
-import java.io.IOException
-import java.net.HttpURLConnection
-import java.net.URL
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.http.HttpStatusCode
+import io.ktor.utils.io.errors.IOException
 
 /** Asks remote server if this version of the app is banned */
-class VersionIsBannedChecker(private val url: String, private val userAgent: String) {
+class VersionIsBannedChecker(
+    private val httpClient: HttpClient,
+    private val url: String,
+    private val userAgent: String
+) {
 
-    fun get(): BannedInfo {
-        var connection: HttpURLConnection? = null
+    suspend fun get(): BannedInfo {
         try {
-            connection = (URL(url).openConnection() as HttpURLConnection)
-            connection.inputStream.bufferedReader().use { reader ->
-                for (line in reader.lineSequence()) {
-                    val text = line.split("\t".toRegex())
-                    if (text[0] == userAgent) {
-                        return IsBanned(if (text.size > 1) text[1] else null)
+            val response = httpClient.get(url)
+            if (response.status == HttpStatusCode.OK) {
```

Comment:
If `expectSuccess` does what I suppose it does, maybe use that instead of checking manually for status code?

In any case, it would be nice if this was consistent in the different downloader/uploader classes.